### PR TITLE
semantic segmentation example

### PR DESCRIPTION
**Pull request by y27choi**
### Linked issue(s):
Fixes KOL-3431

[results example](https://trunk.kolena.io/yoohee/results?modelIds=693&testSuiteId=1435&workflow=Semantic+Segmentation&evaluators=N4IgjALAzArCBcoBsBOKCwCYAMSC%2BeQA)

### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

File: examples/semantic_segmentation/semantic_segmentation/seed_test_run.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+
+def seed_test_run(model_name: str, test_suite_names: List[str]) -> None:
+    def infer(test_sample: TestSample) -> Inference:
+        basename = test_sample.metadata["basename"]
+        locator = f"s3://{BUCKET}/{DATASET}/results/{model_name}/{basename}_person.npy"
+        return Inference(prob=BinaryAsset(locator))
+
+    model = Model(f"{model_name}", infer=infer)
+    print(f"Model: {model}")
```
Comment (munkyshi) [inline examples/semantic_segmentation/semantic_segmentation/seed_test_run.py]: reminder: Remove arbitrary print statements (aren't these already being logged with `verbose=True`?)

File: examples/semantic_segmentation/semantic_segmentation/seed_test_run.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    for test_suite_name in test_suite_names:
+        test_suite = TestSuite.load(test_suite_name)
+        print(f"Test Suite: {test_suite}")
```
Comment (munkyshi) [inline examples/semantic_segmentation/semantic_segmentation/seed_test_run.py]: reminder: Remove arbitrary print statements (aren't these already being logged with `verbose=True`?)

File: examples/semantic_segmentation/semantic_segmentation/seed_test_run.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+import kolena
+from kolena.workflow.asset import BinaryAsset
+from kolena.workflow.test_run import test
+
+
+def seed_test_run(model_name: str, test_suite_names: List[str]) -> None:
+    def infer(test_sample: TestSample) -> Inference:
+        basename = test_sample.metadata["basename"]
+        locator = f"s3://{BUCKET}/{DATASET}/results/{model_name}/{basename}_person.npy"
```
Comment (munkyshi) [inline examples/semantic_segmentation/semantic_segmentation/seed_test_run.py]: Might want to do something to sanitize model name so it's safe for a directory name.

File: examples/semantic_segmentation/semantic_segmentation/workflow.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    """
+
+
+@dataclass(frozen=True)
+class Inference(BaseInference):
+    """Inference type for the Semantic Segmentation workflow."""
+
+    prob: BinaryAsset
+    """
+    The [`BinaryAsset`][kolena.workflow.annotation.BinaryAsset] produced by the model.
```
Comment (munkyshi) [inline examples/semantic_segmentation/semantic_segmentation/workflow.py]: This is expected to be an array of probabilities?

File: examples/semantic_segmentation/README.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+```shell
+$ poetry run python3 semantic_segmentation/seed_test_run.py --help
+usage: seed_test_run.py [-h] [--test_suite TEST_SUITE] [--models MODEL]
+
+optional arguments:
+  -h, --help            show this help message and exit
+  --test_suites TEST_SUITE
+                        Name(s) of the test suite(s) to test.
+  --model MODEL
+                        Name(s) of the model(s) to test.
```
Comment (munkyshi) [inline examples/semantic_segmentation/README.md]:
```suggestion
                        Name of the model to test.
```

File: examples/semantic_segmentation/README.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+Run a script using the `--help` flag for more information:
+
+```shell
+$ poetry run python3 semantic_segmentation/seed_test_run.py --help
+usage: seed_test_run.py [-h] [--test_suite TEST_SUITE] [--models MODEL]
+
+optional arguments:
+  -h, --help            show this help message and exit
+  --test_suites TEST_SUITE
+                        Name(s) of the test suite(s) to test.
```
Comment (munkyshi) [inline examples/semantic_segmentation/README.md]:
```suggestion
  --test_suite TEST_SUITE
                        Name of the test suite to test.
```

Looks like we only accept a single test suite?

File: examples/semantic_segmentation/README.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+First, ensure that the `KOLENA_TOKEN` environment variable is populated in your environment. See our
+[initialization documentation](https://docs.kolena.io/installing-kolena/#initialization) for details.
+
+This project defines two scripts that perform the following operations:
+
+1. [`seed_test_suite.py`](semantic_segmentation/seed_test_suite.py) creates the following test suites:
+
+    - `"coco-stuff-10k"`, containing samples of COCO-Stuff 10K data
+
+2. [`seed_test_run.py`](semantic_segmentation/seed_test_run.py) tests a specified model, e.g. `pspnet_r101-d8_4xb4-40k_coco-stuff10k-512x512`, on the above test suites
```
Comment (munkyshi) [inline examples/semantic_segmentation/README.md]: Do we have results for only this one model? Or do we expect a second one we can use for illustrating comparisons?

File: examples/semantic_segmentation/semantic_segmentation/seed_test_suite.py [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+                annotation_file=record.annotation_file,
+                captions=literal_eval(record.captions),
+                has_person=record.has_person,
+            ),
+        )
+        ground_truth = GroundTruth(mask=SegmentationMask(locator=record.mask, labels=Label.as_label_map()))
+        test_samples.append((test_sample, ground_truth))
+
+    test_case = TestCase(f"complete :: {DATASET}", test_samples=test_samples, reset=True)
+    print(f"Created test case: {test_case}")
```
Comment (munkyshi) [inline examples/semantic_segmentation/semantic_segmentation/seed_test_suite.py]: nit: Does the `test_case` object render nicely to a string? Or is it better to just use the test case name?

File: examples/semantic_segmentation/semantic_segmentation/seed_test_suite.py [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+
+def main(args: Namespace) -> None:
+    kolena.initialize(os.environ["KOLENA_TOKEN"], verbose=True)
+    complete_test_case = seed_complete_test_case(args)
+    test_suite = TestSuite(
+        f"{DATASET}",
+        test_cases=[complete_test_case],
+        reset=True,
+    )
+    print(f"Created test suite: {test_suite}")
```
Comment (munkyshi) [inline examples/semantic_segmentation/semantic_segmentation/seed_test_suite.py]: nit: Does the `test_suite` object render nicely to a string? Or is it better to just use the test suite name?

File: examples/semantic_segmentation/README.md (hunk 7 above)
Comment (y27choi) [reply examples/semantic_segmentation/README.md]: Will add another model results

File: examples/semantic_segmentation/semantic_segmentation/seed_test_run.py (hunk 3 above)
Comment (y27choi) [reply examples/semantic_segmentation/semantic_segmentation/seed_test_run.py]: @sandalns could you take a look at this please?

**Comment by y27choi** [comment]
Updated README.md to include instructions on setting up bucket integration and specifying out bucket.

File: examples/semantic_segmentation/semantic_segmentation/seed_test_run.py (hunk 3 above)
Comment (sandalns) [reply examples/semantic_segmentation/semantic_segmentation/seed_test_run.py]: Will take a look

File: examples/semantic_segmentation/semantic_segmentation/utils.py [hunk 10]
```diff
@@ -67,15 +69,17 @@ def upload_image(locator: str, image: np.ndarray) -> None:
...
-        mask,
-        (shape[0], shape[1]),
-        order=order,
-        mode="constant",
-        anti_aliasing=False,
-        preserve_range=preserve_range,
-    ).astype(np.float32)
+def sanitize_model_name(model_name: str) -> str:
+    """
+    See https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html for object-key name recomandations
```
Comment (munkyshi) [inline examples/semantic_segmentation/semantic_segmentation/utils.py]:
```suggestion
    See https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html for object-key name recommendations
```

**Review by munkyshi — APPROVED** [review]

File: examples/semantic_segmentation/README.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+
+This project defines two scripts that perform the following operations:
+
+1. [`seed_test_suite.py`](semantic_segmentation/seed_test_suite.py) creates the following test suites:
+
+    - `"coco-stuff-10k"`, containing samples of COCO-Stuff 10K data
+
+2. [`seed_test_run.py`](semantic_segmentation/seed_test_run.py) tests a specified model,
+e.g. `pspnet_r101-d8_4xb4-40k_coco-stuff10k-512x512`, `pspnet_r50-d8_4xb4-20k_coco-stuff10k-512x512`, on the above
+test suites
```
Comment (munkyshi) [inline examples/semantic_segmentation/README.md]:
```suggestion
test suite.
```

**Review by munkyshi — APPROVED** [review]

**Pull request merged**